static_report/src/components/shared/TableActionBar.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+  Button,
+  Flex,
+  Text,
+  Icon,
+  ButtonGroup,
+  Tooltip,
+} from '@chakra-ui/react';
+import {
+  FiDatabase,
+  FiCreditCard,
+  FiBookOpen,
+  FiAlertCircle,
+} from 'react-icons/fi';
+
+export type TableActionBarView = 'schema' | 'summary';
+
+type Props = {
+  currentView: 'schema' | 'summary';
+  toggleView: (view: TableActionBarView) => void;
+  sourceName: string;
+  sourceType: string;
+  inTransition?: boolean;
+};
+
+export function TableActionBar({
+  currentView,
+  toggleView,
+  sourceName,
+  sourceType,
+  inTransition = false,
```

Comment:
suggestion: `isPending` to designate a `boolean` of a state, similar to `Promise`'s `pending` in-flight property. This is to narrow the meaning further.

I think `transition` can be seen as an actual DOM-manipulation/animation type of way, where it can mean many other things.

Reply:
Will be removed.